buildchow/templates/component-template.yaml

```diff
@@ -0,0 +1,137 @@
+            - JOB_NAME=${{JOB_NAME}}
+            - BUILD_ID=${{BUILD_ID}}
+            - BUILD_NUMBER=${{BUILD_NUMBER}}
+            - skipPrivateRequirements=false
+            - jbosstools.test.jre.5="${{NATIVE_TOOLS}}${{SEP}}${{JAVA15}}" 
+            - jbosstools.test.jre.6="${{NATIVE_TOOLS}}${{SEP}}${{JAVA16}}"
+            - jbosstools.test.jre.7="${{NATIVE_TOOLS}}${{SEP}}${{JAVA17}}"
+            - jbosstools.test.jre.8="${{NATIVE_TOOLS}}${{SEP}}${{JAVA18}}"
+          settings: "{maven-settings}"
+          private-repository: true
+          java-opts: 
+            - "{jvm-memory}"
+          pom: ${{WORKSPACE}}/sources/pom.xml
+      - maven-target:
+          goals: "{ci-maven-flags} org.apache.maven.plugins:maven-dependency-plugin:2.9:unpack" #TODO: should it have all MAVEN_FLAGS ?
+          maven-version: "{default-maven-version}"
+          properties:
+            - artifact=org.jboss.tools.releng:jbosstools-releng-publish:4.3.0.Beta1-SNAPSHOT:zip
+            - outputDirectory=${{WORKSPACE}}/sources
+            - trimVersion=true
+            - mdep.stripClassifier=true
+            - mdep.stripVersion=true
+          private-repository: true
+           # TODO: this skiprevision check or at leat the mvn call should not be done in shell. Move to mvn ?
+      - shell: |
+               if [[ ${{skipRevisionCheckWhenPublishing}} == "true" ]] || [[ $(. ${{WORKSPACE}}/sources/util/checkLatestPublishedSHA.sh -s ${{WORKSPACE}}/sources/site/target/repository -t http://download.jboss.org/jbosstools/mars/snapshots/builds/${{JOB_NAME}}/latest/all/repo/) == "true" ]]; then
+                 cd ${{WORKSPACE}}/sources/site
+                 M2_HOME=/qa/tools/opt/apache-maven-3.2.5/
+                 $M2_HOME/bin/mvn deploy -Pdeploy-to-jboss.org -Dmaven.repo.local=$WORKSPACE/.repository ${{MAVEN_FLAGS}}
+               else
+                 echo "Publish cancelled (nothing to do). Skip this check with skipRevisionCheckWhenPublishing=true"
+                 BUILD_DESCRIPTION="NOT PUBLISHED: UNCHANGED"
+               fi
+
+               echo "Available JREs for testing:"
+               echo "jbosstools.test.jre.5=${{NATIVE_TOOLS}}${{SEP}}${{JAVA15}}"
+               echo "jbosstools.test.jre.6=${{NATIVE_TOOLS}}${{SEP}}${{JAVA16}}"
+               echo "jbosstools.test.jre.7=${{NATIVE_TOOLS}}${{SEP}}${{JAVA17}}"
+               echo "jbosstools.test.jre.8=${{NATIVE_TOOLS}}${{SEP}}${{JAVA18}}"
+      - maven-target:
+          goals: verify ${{MAVEN_FLAGS}} -DTARGET_PLATFORM_VERSION=${{TARGET_PLATFORM_VERSION_MAXIMUM}}
+          maven-version: "{default-maven-version}"
+          java-opts: 
+            - "{jvm-memory}"
+          pom: "${{WORKSPACE}}/sources/{tests-folder}/pom.xml"
+          private-repository: true
+          properties:
+            *component-maven-properties
+    publishers:
+      - archive:
+          artifacts: >
+            results/*/all/*Update*.zip,sources/*/target/jacoco.exec,
```

Comment:
I would use *.zip here instead of *Update*.zip; at some point we may have source zips for the projects too, not just update sites.